qemu/tests/virtio_console.py

```diff
@@ -570,12 +571,154 @@ def test_loopback():
+                                   'name': property['name'],
+                                   'bus': property['bus']}.items():
+                    portdev.set_param(key, value)
+                (out, ver_out) = vm.devices.simple_hotplug(portdev, vm.monitor)
+                if not ver_out:
+                    test.error("Error occured when plug port %s." % property['name'])
+
+            def _pci_unplug(bus):
+                device = get_device_by_id(vm, str(bus).split('.')[0])
+                if not device:
+                    test.error("No bus %s in vm." % bus)
+                bus_property = dict(id=device.get_param("id"))
+                err = ""
+                out = device.unplug(vm.monitor)
+                if out:
+                    err += "\nHotplug monitor output: " + out
+                time.sleep(2)
+                ver_out = device.verify_unplug(out, vm.monitor)
+                deathline = time.time() + 5
+                while time.time() < deathline:
+                    if ver_out is True:
+                        device.unplug_hook()
+                        vm.devices.remove(device, True)
+                        break  # Unplugged successfully
+                    elif "unknown command" in ver_out:
+                        logging.warn(
+                                "Can't verify if the device was removed "
+                                "because info qtree is unsupported. Waiting"
+                                "10s and hoping it's sufficient.")
+                        time.sleep(10)
+                        break
+                    else:
+                        # When kernel doesn't answer to ACPI unplug event, pci is
+                        # not unplugged. We have to resend the unplug cmd.
+                        out = device.unplug(vm.monitor)
+                    time.sleep(2)
+                    ver_out = device.verify_unplug(out, vm.monitor)
+                else:
+                    logging.error(vm.monitor.info("qtree"))
+                    if err:
+                        logging.error(
+                                "Device not unplugged after 20s.\nHotplug "
+                                "errors:%s", err)
+                    test.fail("Device not unplugged after 20s. ")
+
+                if err != "":
+                    logging.error(vm.monitor.info("qtree"))
+                    test.fail(
+                            "Error occurred while hotpluging virtio-"
+                            "pci. Monitor output:%s" % err)
+                return bus_property
+
+            def _pci_plug(property):
+                bus = qdevices.QDevice("virtio-serial-pci")
+                bus.set_param('id', property['id'])
+                (out, ver_out) = vm.devices.simple_hotplug(bus, vm.monitor)
+                if not ver_out:
+                    test.error("Error occured when plug bus. out: %s", out)
+
+            send_prop = _port_unplug(0)
+            time.sleep(intr_time)
```

Comment:
better put `time.sleep(intr_time)` into functions like `_port_unplug`

Reply:
Yes, you are right, will update it, thanks!